frontend/src/components/ColorLegend.svelte

```diff
@@ -1,28 +1,96 @@
 <script lang="ts">
-    export let colors;
+    export let styles;
```

Comment:
I am not sure about moving all this here
It makes the overall architecture a bit confusing since it is here now that the graph is created whereas I would expect to have styles only. What is the purpose of the G6Graph.svelte file?